### PR TITLE
[CirrusCI] Matrix Cron Task and Manual Task

### PR DESCRIPTION
**Pull request by evertonlperes**
### Changes
1. Implemented cron/manual trigger matrix in order to enable automated CRON Tasks along with manual trigger for PRs and Merges.

Remark:
The Cirrus CI pipelines is activated only on my forked [repository](https://github.com/evertonlperes/rancher-desktop) for now. @gunamata and I decided to keep it enabled only on my forked, to collect data and ensure the pipeline is stable enough and then, enable at the upstream repository.

### CODE REVIEW

**Review by mook-as — CHANGES_REQUESTED** [review]
> As mentioned on Slack — the description of the PR is unclear; we need a better one (and ideally, matching commit message change) to explain why this is desired.

Yep, I'll modify it. Thanks for the heads up.

File: .cirrus.yml [hunk 1]
```diff
@@ -1,5 +1,14 @@
 task:
-  trigger_type: manual
+  # On Cirrus CI, cron tasks does not start if it has
+  # manual trigger type enabled - expected behaviour
+  # see: https://github.com/cirruslabs/cirrus-ci-docs/discussions/949#discussioncomment-1853964
+  matrix:
+    # Keep manual trigger for all pushes (prs and merges)
+    - trigger_type: manual
+      only_if: $CIRRUS_CRON == ''
+    # Only trigger task automatically if it's a CRON and main branch
```
Comment (mook-as) [inline .cirrus.yml:9]:
```suggestion
    # Only trigger task automatically if it's a cron and main branch
```
`cron` is not an initialism, as far as I know, and `man cron` doesn't indicate it should be upper case.

**Review by mook-as — CHANGES_REQUESTED** [review]
The commits can use some rewording and squashing to be clear what we're trying to do here.  Suggestion:

```
Ensure that cirrus automatically triggers tasks when invoked by cron.

On Cirrus CI, a CI job configured for manual triggering isn't triggered
even if it matches a cron expression set in the repository options.  By
making it `trigger_type: automatic` when invoked via cron, we can
ensure the correct thing happens, and that the jobs are triggered
either via cron or via a manual trigger (but not on push).
```

**Comment by evertonlperes** [comment]
Closing this to start a fresh PR with better commit messages and description.
Just a few changes and it won't hunt start a new one along with Mark's comments and proper messages/naming. Thanks @mook-as for reviewing it.

**Pull request closed without merging**